File: programming/project-learning/php/maths/russian-peasant-long-multiplication.php
<?php
// Prompt the user to enter the first operand
echo "Enter op1: ";
// Read the first operand from user input, trim any whitespace, and convert it to an integer
$op1 = intval(trim(fgets(STDIN)));

// Prompt the user to enter the second operand
echo "Enter op2: ";
// Read the second operand from user input, trim any whitespace, and convert it to an integer
$op2 = intval(trim(fgets(STDIN)));

// Initialize the answer to 0
$ans = 0;

// Loop until the first operand becomes 0
while ($op1 > 0) {
    // If the first operand is odd, add the second operand to the answer
    if ($op1 % 2 != 0) {
        $ans += $op2;
        // Print the current value of the answer
        echo "$ans\n";
    }

    // Halve the first operand (integer division by 2)
    $op1 = intdiv($op1, 2);
    // Double the second operand
    $op2 *= 2;

    // Print the current values of the operands
    echo "$op1 $op2\n";
}

// Print the final answer
echo "The answer is: $ans\n";
?>

File: programming/project-learning/r/maths/russian-peasant-long-multiplication.r
ans <- 0  # Initialize the answer to 0

op1 <- as.integer(readline(prompt = "Enter op1: "))  # Prompt the user to enter the first operand and convert it to an integer
op2 <- as.integer(readline(prompt = "Enter op2: "))  # Prompt the user to enter the second operand and convert it to an integer

while (op1 > 0) {  # Continue the loop while the first operand is greater than 0
    if (op1 %% 2 != 0) {  # Check if the first operand is odd
        ans <- ans + op2  # If odd, add the second operand to the answer
        print(ans)  # Print the current value of the answer
    }

    op1 <- op1 %/% 2  # Divide the first operand by 2 (integer division)
    op2 <- op2 * 2  # Multiply the second operand by 2

    print(c(op1, op2))  # Print the current values of the operands
}

cat("The answer is: ", ans, "\n")  # Print the final answer